cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -1342,7 +1342,7 @@ AsofJoinNode::AsofJoinNode(ExecPlan* plan, NodeVector inputs,
       must_hash_(must_hash),
       may_rehash_(may_rehash),
       tolerance_(tolerance),
-      backpressure_counter_(0),
+      backpressure_counter_(1),
```

Comment:
@westonpace Not sure why this would cause back pressure to propagated to one of the inputs, can you shed some light?

Reply:
nvm - saw in the PR description